functions/go/set-namespace/transformer/namespace.go

```diff
@@ -98,45 +71,152 @@ func (p *NamespaceTransformer) Config(o *fn.KubeObject) error {
 	return nil
 }
 
+// Transform replace the existing Namespace object, namespace-scoped resources' `metadata.name` and
+// other namespace reference fields to the new value.
 func (p *NamespaceTransformer) Transform(objects []*fn.KubeObject) {
 	p.SetDependsOnMap(objects)
 	for _, o := range objects {
 		switch {
+		// Skip local resource which `kpt live apply` skips.
+		case o.IsLocalConfig():
+			continue
 		case o.IsGVK("v1", "Namespace"):
-			o.SetName(p.Namespace)
+			o.SetName(p.NewNamespace)
```

Comment:
I do think we want to check whether the old namespace matches.  If we feel the semantics of this function are sufficiently different, we might just want to make a new function and deprecate this one.